comp/core/workloadmeta/collectors/internal/host/host.go

```diff
@@ -57,53 +63,59 @@ func (c *collector) Start(ctx context.Context, store workloadmeta.Component) err
 	if duration <= time.Minute {
 		log.Debugf("Tags are checked for expiration once per minute. expected_tags_duration should be at least one minute and in minute intervals.")
 	}
-	tags := hostMetadataUtils.Get(ctx, false, c.config).System
-	log.Debugf("Adding host tags to metrics for %v : %v", duration, tags)
 
-	store.Notify([]workloadmeta.CollectorEvent{
-		{
-			Type:   workloadmeta.EventTypeSet,
-			Source: workloadmeta.SourceNodeOrchestrator,
-			Entity: makeEntity(tags),
-		},
-	})
-
-	go func() {
-		select {
-		case <-ctx.Done():
-			return
-
-		case <-c.clock.After(duration):
-			store.Notify([]workloadmeta.CollectorEvent{
-				{
-					Type:   workloadmeta.EventTypeSet,
-					Source: workloadmeta.SourceNodeOrchestrator,
-					Entity: makeEntity([]string{}),
-				},
-			})
-		}
-	}()
+	log.Debugf("Adding host tags to metrics for %v", duration)
+	c.timeoutTimer = c.clock.Timer(duration)
+
 	return nil
 }
 
-func makeEntity(tags []string) *workloadmeta.HostTags {
-	return &workloadmeta.HostTags{
-		EntityID: workloadmeta.EntityID{
-			Kind: workloadmeta.KindHost,
-			ID:   "host-tags",
-		},
-		HostTags: tags,
+func (c *collector) Pull(ctx context.Context) error {
+	// Feature is disbled or timeout has previously occurred
+	if c.timeoutTimer == nil {
+		return nil
+	}
+
+	// Timeout reached - expire any host tags in the store
+	if c.isTimedOut() {
+		c.store.Notify(makeEvent([]string{}))
+		return nil
 	}
-}
 
-func (c *collector) Pull(_ context.Context) error {
+	tags := hostMetadataUtils.Get(ctx, false, c.config).System
+	c.store.Notify(makeEvent(tags))
 	return nil
 }
 
 func (c *collector) GetID() string {
-	return "host"
+	return id
 }
 
 func (c *collector) GetTargetCatalog() workloadmeta.AgentType {
 	return c.catalog
 }
+
+func (c *collector) isTimedOut() bool {
```

Comment:
nit: function name is a "boolean" looking name `isTimedOut` but mutates state. I proposed `checkTimeoutTimer` or something.